clientconn.go

```diff
@@ -143,6 +143,7 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 		firstResolveEvent: grpcsync.NewEvent(),
 	}
 	cc.retryThrottler.Store((*retryThrottler)(nil))
+	cc.safeConfigSelector.UpdateConfigSelector(&defaultConfigSelector{nil})
```

Comment:
It seems #4343 can reproduce this bug ~reliably.
Did you look more into that and figure out how it happened?
Maybe we can make a test case from it.

Reply:
The logs from #4343 are how I figured out what's happening here.  What's happening there is the resolver is reporting an error while the clientconn is being shut down.  It seems hard to stimulate it reliably since: 1. the window is very small between steps 3-4, and even if you hit it the odds of (6) happening also seems low.